File: doc-examples/jdbc-example-records-java/src/main/java/example/AccountRepository.java
package example;

import io.micronaut.data.annotation.Id;
import io.micronaut.data.annotation.Query;
import io.micronaut.data.jdbc.annotation.JdbcRepository;
import io.micronaut.data.model.query.builder.sql.Dialect;
import io.micronaut.data.repository.CrudRepository;

// tag::reservable[]
@JdbcRepository(dialect = Dialect.ORACLE)
public interface AccountRepository extends CrudRepository<Account, Long> {

    int reserveIncrementBalance(@Id Long id, Long balance);
}
// end::reservable[]

// tag::reservable-raw-query[]
@JdbcRepository(dialect = Dialect.ORACLE)
interface AccountRawQueryRepository extends CrudRepository<Account, Long> {

    @Query("UPDATE \"ACCOUNT\" SET \"BALANCE\" = \"BALANCE\" + :amount WHERE \"ID\" = :id")
    int reserveBalance(Long id, Long amount);
}
// end::reservable-raw-query[]

File: doc-examples/jdbc-example-records-python/src/main/python/example/AccountRepository.py
from typing import Annotated

from micronaut.data.annotation import Id, Query
from micronaut.data.jdbc.annotation import JdbcRepository
from micronaut.data.model.query.builder.sql import Dialect
from micronaut.data.repository import CrudRepository

from example.Account import Account


# tag::reservable[]
@JdbcRepository(dialect=Dialect.ORACLE)
class AccountRepository(CrudRepository[Account, int]):

    def reserveIncrementBalance(self, id: Annotated[int, Id], balance: int) -> int: ...
# end::reservable[]


# tag::reservable-raw-query[]
@JdbcRepository(dialect=Dialect.ORACLE)
class AccountRawQueryRepository(CrudRepository[Account, int]):

    @Query('UPDATE "ACCOUNT" SET "BALANCE" = "BALANCE" + :amount WHERE "ID" = :id')
    def reserveBalance(self, id: int, amount: int) -> int: ...
# end::reservable-raw-query[]
